.github/workflows/build.yml

```
@@ -8,33 +8,26 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  # ruby_version 共通化のためのジョブ
+  # 参考: https://docs.github.com/en/actions/how-tos/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#example-using-an-output-to-define-two-matrices
+  define-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      ruby_versions: ${{ steps.ruby_versions.outputs.ruby_versions }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Define Ruby versions
+        id: ruby_versions
+        run: echo "ruby_versions=$(cat ruby_versions.json | jq -c)" >> "$GITHUB_OUTPUT" # 1行にする必要があるので "jq -c" を実施
```

好み
標準入力でデータを流すだけならリダイレクトでもいけそう
```suggestion
        run: echo "ruby_versions=$(jq -c <ruby_versions.json)" >> "$GITHUB_OUTPUT" # 1行にする必要があるので "jq -c" を実施
```